runtime/src/iree/schemas/cpu_data.h

```diff
@@ -70,7 +70,31 @@ enum iree_cpu_data_field_0_e {
   IREE_CPU_FEATURE_BIT_NAME(arch, field_index, bit_name) = 1ull << bit_pos,
 #include "iree/schemas/cpu_feature_bits.inl"
 #undef IREE_CPU_FEATURE_BIT
+};
+
+// Struct to capture tuple of llvm feature-name to bit pattern used to represent
+// it.
+struct iree_llvm_name_and_bit_pattern_t {
```

Comment:
Please don't touch this `cpu_data.h` file and instead move this to `DispatchABI.cpp`, where this is currently used.

These helpers and the logic in `DispatchABI.cpp` are assuming that only CPU data field 0 is used. This is only incidentally true at the moment.  I can fix that in a follow-up.  It's best if all this interrelated code is localized in `DispatchABI.cpp` so my fixup is localized there.

Reply:
I need there to be a single source of truth, so that the compiler can see the same logic as the microkernel library. Otherwise things won't inline as they should. I am happy to move it anywhere, but what pieces do you want me to move?
Agreed that the fact that only cpu0 is used is incidental, but how do we make this handshake work. This is effectively reflection. To fix this we will need to use helper methods in the microkernel library. Wondering if we can do all that as a follow up

Reply:
`iree_llvm_name_and_bit_pattern_list` is defined in `cpu_data.h` and referenced only in `DispatchABI.cpp`. So it should be trivial to move it to `DispatchABI.cpp` ?